File: support_tools/swigwin-3.0.12/Examples/test-suite/csharp/nested_workaround_runme.cs
using System;
using nested_workaroundNamespace;
#pragma warning disable 219

public class runme {
  static void Main() {
    {
      Inner inner = new Inner(5);
      Outer outer = new Outer();
      Inner newInner = outer.doubleInnerValue(inner);
      if (newInner.getValue() != 10)
        throw new Exception("inner failed");
    }

    {
      Outer outer = new Outer();
      Inner inner = outer.createInner(3);
      Inner newInner = outer.doubleInnerValue(inner);
      if (outer.getInnerValue(newInner) != 6)
        throw new Exception("inner failed");
    }
  }
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/nested_workaround_runme.lua
require("import")	-- the import fn
import("nested_workaround")	-- import lib
nw=nested_workaround

-- catch "undefined" global variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

i1 = nw.Inner(5)
assert(i1:getValue() == 5)
i1:setValue(7)
assert(i1:getValue() == 7)

o1 = nw.Outer()
i2 = o1:createInner(9)
assert(i2:getValue() == 9)
i2:setValue(11)
assert(o1:getInnerValue(i2) == 11)

i3 = o1:doubleInnerValue(i2)
assert(i3:getValue() == 22)
